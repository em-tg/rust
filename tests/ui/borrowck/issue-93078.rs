//@ check-pass
trait Modify {
    fn modify(&mut self) ;
}

impl<T> Modify for T  {
    fn modify(&mut self)  {}
}

trait Foo {
    fn mute(&mut self) {
        self.modify(); //~ WARNING cannot borrow `self` as mutable
    }
}

fn main() {}
