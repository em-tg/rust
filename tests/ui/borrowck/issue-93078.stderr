warning[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> $DIR/issue-93078.rs:12:9
   |
LL |         self.modify();
   |         ^^^^ cannot borrow as mutable
   |
   = note: as `Self` may be unsized, this call attempts to take `&mut &mut self`
   = note: however, `&mut self` expands to `self: &mut Self`, therefore `self` cannot be borrowed mutably
   = note: `#[warn(mut_non_mut)]` on by default

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0596`.
